src/main/webapp/app/core/util/alert.service.ts

```diff
@@ -5,29 +5,35 @@ import { translationNotFoundMessage } from 'app/core/config/translation.config';
 import { EventManager, EventWithContent } from 'app/core/util/event-manager.service';
 import { AlertError } from 'app/shared/alert/alert-error.model';
 import { Subscription } from 'rxjs';
+import { captureException } from '@sentry/browser';
 
 export type AlertType = 'success' | 'danger' | 'warning' | 'info';
 
-export interface AlertBase {
+interface AlertBaseInternal {
     type: AlertType;
     message?: string;
-    translationKey?: string;
-    translationParams?: { [key: string]: unknown };
     timeout?: number;
-    action?: { label: string; callback: (alert: Alert) => void };
+    action?: { readonly label: string; readonly callback: (alert: Alert) => void };
     onClose?: (alert: Alert) => void;
     dismissible?: boolean;
 }
 
-export interface Alert extends AlertBase {
-    id: number;
-    close: () => void;
+export interface AlertCreationProperties extends Readonly<AlertBaseInternal> {
```

Comment:
If this extends the Base readonly, how can the values actually be filled?

Reply:
At object creation, simply like this (example). Only field overwrites after the object has been created are not permitted. However, we probably don't even need to make it readonly as we're using (at least atm) a defensive copy, and even if we didn't, users could just init the object as ``any`` and cast to ``AlertCreationProperties`` later... So I think we can just not make it readonly
```js
this.alertService.addAlert({
    type: 'info',
    message: 'My default message',
    translationKey: 'artemisApp.exampleTranslation'
    timeout: 9000
 });
 ```